plugins/opener/guest-js/init.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+import { invoke } from '@tauri-apps/api/core'
+
+// open <a href="..."> links with the API
+function openLinks(): void {
+  document.querySelector('body')?.addEventListener('click', function (e) {
+    let target: HTMLElement | null = e.target as HTMLElement
+    while (target) {
+      if (target.matches('a')) {
+        const t = target as HTMLAnchorElement
+        if (
+          t.href !== '' &&
+          ['http://', 'https://', 'mailto:', 'tel:'].some((v) =>
+            t.href.startsWith(v)
+          ) &&
+          t.target === '_blank'
+        ) {
+          void invoke('plugin:opener|open', {
+            path: t.href
+          })
+          e.preventDefault()
+        }
+        break
+      }
+      target = target.parentElement
+    }
+  })
+}
+
+if (
+  document.readyState === 'complete' ||
+  document.readyState === 'interactive'
+) {
+  openLinks()
+} else {
+  window.addEventListener('DOMContentLoaded', openLinks, true)
+}
```

Comment:
What about we just listen `click` events on `window`?

Reply:
Also, about finding if an anchor tag is clicked, what about we use something like this https://github.com/solidjs/solid-router/blob/main/src/data/events.ts

Reply:
> What about we just listen `click` events on `window`?

let's keep it as is for now, I don't know if this was done on purpose or not and I don't want to break anything 

> Also, about finding if an anchor tag is clicked, what about we use something like this [solidjs/solid-router@`main`/src/data/events.ts](https://github.com/solidjs/solid-router/blob/main/src/data/events.ts?rgh-link-date=2024-11-12T02%3A39%3A26Z)

sure, but is our